### ISSUE TITLE
ROS Humble - Failed to resolve symbol

### ISSUE THREAD

**Issue by jasperdekeuk** [eclipse-ecal/rmw_ecal#56]
Big fan of Ecal!
Now trying to do the integration in ROS2, but have not managed to get it working.

# Setup

OS: 

- Ubuntu 22.04.1 LTS (jammy)

Ecal: 

- Rolling release (5.11.1)

ROS:

- Humble Hawksbill

# What is working

- Ecal on it's own is working fine (both local and cloud)
- ROS2 demo nodes are working fine (monitor with rosBoard)

# Problem

- Integrating Ecal as ros RMW
- Error when starting the demo_node_cpp talker whith rmw_ecal_dynamic_cpp and also rmw_ecal_proto_cpp

> 'failed to resolve symbol 'rmw_subscription_set_content_filter' in shared library '/home/user/ros2_ecal_rmw/install/rmw_ecal_dynamic_cpp/lib/librmw_ecal_dynamic_cpp.so', at ./src/functions.cpp:152'

- this goes on for basically all symbols in librmw_ecal_dynamic_cpp.so. Same goes for librmw_ecal_proto_cpp when setting that rmw

# Steps to reproduce

```shell
source /opt/ros/humble/setup.bash
mkdir -p ~/ros2_ecal_ws/src && cd ~/ros2_ecal_ws/src
git clone https://github.com/eclipse-ecal/rmw_ecal.git
git clone https://github.com/jasperdekeuk/rosidl_typesupport_protobuf.git
cd ..
colcon build
source install/setup.bash
RMW_IMPLEMENTATION=rmw_ecal_proto_cpp ros2 run demo_nodes_cpp talker
```

Not much experience in ROS, so might be missing something. For some reason I also have to call "colcon build" 2x to make sure everything is properly build.

**Comment by FlorianReimold** [comment]
Thanks! maybe you can create a PR when you have a working solution? That would help us out a lot!

**Comment by jasperdekeuk** [comment]
My solution makes it buildable, but unfortunately not yet runnable. The errors from the issue are with the updated version. 
Seems to be a ROS problem though.

**Comment by jasperdekeuk** [comment]
This problem seems to be similar, on another ROS rmw (and seems to be fixt)
-> https://github.com/ros2/rmw_fastrtps/issues/314

**Comment by FlorianReimold** [comment]
Fixed in #58
